CMakeLists.txt

```diff
@@ -243,6 +245,7 @@ if(NOT APPLE OR NOT ${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm64")
 endif()
 
 option(IREE_HAL_DRIVER_CUDA "Enables the 'cuda' runtime HAL driver" ${IREE_HAL_DRIVER_CUDA_DEFAULT})
+option(IREE_HAL_DRIVER_CUDA2 "Enables the 'cuda2' runtime HAL driver" ${IREE_HAL_DRIVER_CUDA2_DEFAULT})
```

Comment:
Can we come up with a plan for switching from the old `cuda` to the new `cuda2`? Maybe post an RFC and an update to https://github.com/openxla/iree/issues/13245, and could link to one or both of those here.

Some ideas:

* Drive to feature complete
  * All CTS tests and execution tests we care about pass using `cuda2`
  * Figure out the story for flags like on https://github.com/openxla/iree/issues/13984
* Switch default in-repo
  * All in-repo benchmarks using `cuda2`, without egregious regressions
* Delete (old) `cuda` and rename `cuda2` to `cuda`

Ideally we'd have a timeline in mind so we don't stay in this state of having two similar paths for long.

Reply:
SG. I've commented on #13245 with https://github.com/openxla/iree/issues/13245#issuecomment-1684210183. I'll send out a mail later too.

Reply:
Hard for me to give a precise timeline though; given I'm multiplexing over quite a few work streams. But yeah agreed we don't want to be in a hybrid state for too long; I'm hoping to get feature parity down soon.